Realm/ObjectServerTests/RealmServer.swift

```diff
@@ -0,0 +1,786 @@
+        /// This allows us to call a give endpoint dynamically with loose typing.
+        @dynamicMemberLookup
+        struct AdminEndpoint {
+            /// The access token of the authenticated user
+            var accessToken: String
+            /// The group id associated with the authenticated user
+            var groupId: String
+            /// The endpoint url. This will be appending to dynamically by appending the dynamic member called
+            /// as if it were a path.
+            var url: URL
+            
+            /**
+             Append the given member to the path. E.g., if the current URL is set to
+            http://localhost:9090/api/admin/v3.0/groups/groupId/apps/appId
+             and you currently have a:
+             ```
+             var app: AdminEndpoint
+             ```
+             you can fetch a list of all services by calling
+             ```
+             app.services.get()
+             ```
+             */
+            subscript(dynamicMember member: String) -> AdminEndpoint {
+                let pattern = "([a-z0-9])([A-Z])"
+
+                let regex = try? NSRegularExpression(pattern: pattern, options: [])
+                let range = NSRange(location: 0, length: member.count)
+                let snakeCaseMember = regex?.stringByReplacingMatches(in: member,
+                                                                      options: [],
+                                                                      range: range,
+                                                                      withTemplate: "$1_$2").lowercased()
+                return AdminEndpoint(accessToken: accessToken,
+                                     groupId: groupId,
+                                     url: url.appendingPathComponent(snakeCaseMember!))
+            }
+            
+            /**
+             Append the given id to the path. E.g., if the current URL is set to
+             http://localhost:9090/api/admin/v3.0/groups/groupId/apps/
+              and you currently have a:
+              ```
+              var apps: AdminEndpoint
+              var appId: String
+              ```
+              you can fetch the app from its appId with
+              ```
+              apps[appId].get()
+              ```
+             */
+            subscript(_ id: String) -> AdminEndpoint {
+                get {
+                    return AdminEndpoint(accessToken: accessToken,
+                                         groupId: groupId,
+                                         url: url.appendingPathComponent(id))
+                }
+            }
+
+            private func request(httpMethod: String, data: Any? = nil, completionHandler: @escaping (Any?, Error?) -> ()) {
+                var request = URLRequest(url: url)
+                request.httpMethod = httpMethod
+                request.allHTTPHeaderFields = [
+                    "Authorization": "Bearer \(String(describing: accessToken))",
```

Comment:
Why not `"Bearer \(accessToken)"`?

Reply:
This was likely nullable before and is an artefact of that. Will change.